packages/angular/src/errorhandler.ts

```diff
@@ -50,6 +50,35 @@ function extractHttpModuleError(error: HttpErrorResponse): string | Error {
   return error.message;
 }
 
+function isObject(value: unknown): value is Record<string, unknown> {
+  return value !== null && typeof value === 'object';
+}
+
+function hasOwnProperty<ObjectType extends Record<string, unknown>, PropertyType extends PropertyKey>(
+  object: ObjectType,
+  propertyName: PropertyType,
+): object is ObjectType & Record<PropertyType, unknown> {
+  return Object.prototype.hasOwnProperty.call(object, propertyName);
+}
+
+function hasErrorName(value: Record<string, unknown>): value is Pick<Error, 'name'> {
+  return hasOwnProperty(value, 'name') && typeof value.name === 'string';
+}
+
+function hasErrorMessage(value: Record<string, unknown>): value is Pick<Error, 'message'> {
+  return hasOwnProperty(value, 'message') && typeof value.message === 'string';
+}
+
+function hasErrorStack(value: Record<string, unknown>): value is Pick<Error, 'stack'> {
+  return !hasOwnProperty(value, 'stack') || typeof value.stack === 'string';
+}
+
+function isErrorOrErrorLikeObject(value: unknown): value is Error {
+  return (
+    value instanceof Error || (isObject(value) && hasErrorName(value) && hasErrorMessage(value) && hasErrorStack(value))
```

Comment:
of course this could all be inlined, but I find a lot more readable that way

Reply:
It's very readable but we also have to consider the bundle size impact here. Hence my question if `hasOwnProperty` is strictly necessary. 

Maybe, we could also inline the `isObject` check?

Reply:
I'm not very familiar with how the code impacts the bundle size, is there a way I could check the result so that I can check different things to see if there is still a way to keep it readable without increasing the size too much?